test/group.lean

```diff
@@ -0,0 +1,59 @@
+import tactic.group
+
+variables {G : Type} [group G]
+
+example (a b c d : G) : c *(a*b)*(b⁻¹*a⁻¹)*c = c*c :=
+by group
+
+example (a b c d : G) : (b*c⁻¹)*c *(a*b)*(b⁻¹*a⁻¹)*c = b*c :=
+by group
+
+example (a b c d : G) : c⁻¹*(b*c⁻¹)*c *(a*b)*(b⁻¹*a⁻¹*b⁻¹)*c = 1 :=
+by group
+
+
+def commutator {G} [group G] : G → G → G := λ g h, g * h * g⁻¹ * h⁻¹
+
+def commutator3 {G} [group G] : G → G → G → G := λ g h k, commutator (commutator g h) k
+
+-- The following is known as the Hall-Witt identity, see e.g. https://en.wikipedia.org/wiki/Three_subgroups_lemma#Proof_and_the_Hall%E2%80%93Witt_identity
+example (g h k : G) :
+  g * (commutator3 g⁻¹ h k) * g⁻¹ * k * (commutator3 k⁻¹ g h) * k⁻¹ * h * (commutator3 h⁻¹ k g) * h⁻¹ = 1 :=
+by { dsimp [commutator3, commutator], group }
```

Comment:
Is this identity important enough to be in mathlib proper?

Reply:
I don't think so. It was suggested as a test, but it wouldn't make much sense, even if you formalize the whole wikipedia page linked above.

Reply:
I think you're right. If anyone has a use for this in the future they can extract it then.